### ISSUE TITLE
The /docs endpoint won't work with spring boot 1.4.2.RELEASE

### ISSUE THREAD

**Issue by eacdy** [spring-projects/spring-boot#7489]
I've included the two depedencies:
```
    <dependency>
      <groupId>org.springframework.boot</groupId>
      <artifactId>spring-boot-starter-actuator</artifactId>
    </dependency>

    <dependency>
      <groupId>org.springframework.boot</groupId>
      <artifactId>spring-boot-actuator-docs</artifactId>
    </dependency>
```

And then with Spring Boot 1.4.2 the /doc endpoint returns me 404. But When I use Spring Boot 1.4.1 and below, It works.

**Comment by eacdy** [comment]
I've include the two depedencies:
```
    <dependency>
      <groupId>org.springframework.boot</groupId>
      <artifactId>spring-boot-starter-actuator</artifactId>
    </dependency>

    <dependency>
      <groupId>org.springframework.boot</groupId>
      <artifactId>spring-boot-actuator-docs</artifactId>
    </dependency>
```


And then with Spring Boot 1.4.2 the /doc endpoint returns me 404. But When I use Spring Boot 1.4.1 and below, It works.

**Comment by eacdy** [comment]
I've found the reason.
When I use `spring-boot-starter-actuator` 1.4.2, It doesn't contain the dependency
```
    <dependency>
      <groupId>org.springframework.hateoas</groupId>
      <artifactId>spring-hateoas</artifactId>
    </dependency>
```


But 1.4.1 has already contain this denpency.

** I suggest the Spring Boot document should mention this change rather than just the words below. **


Displays documentation, including example requests and responses, for the Actuator’s endpoints. Requires spring-boot-actuator-docs to be on the classpath.

**Comment by snicoll** [comment]
I've introduced that regression while polishing #7215

**Comment by snicoll** [comment]
@eacdy this is fixed, you can workaround with 1.4.2 by adding the following dependency to your project:

```xml
   <dependency>
      <groupId>org.springframework.hateoas</groupId>
      <artifactId>spring-hateoas</artifactId>
    </dependency>
```

**Comment by eacdy** [comment]
@snicoll 
thank you.I've found that. 
I suggest the Spring Boot document should mention this change rather than just the words below.

Displays documentation, including example requests and responses, for the Actuator’s endpoints. Requires spring-boot-actuator-docs to be on the classpath.
